src/fprime/fbuild/interaction.py

```diff
@@ -69,41 +74,91 @@ def run_impl(deployment: Path, path: Path, platform: str, verbose: bool):
         lines = lines[11:]  # Remove old header
         with open(dest, "a") as file_handle:
             file_handle.write("".join(lines))
-    removals = [path / "mod.mk", path / "Makefile", Path(hpp_src), Path(cpp_src)]
-    for removal in removals:
-        os.remove(removal)
     return True
 
 
 def add_to_cmake(list_file: Path, comp_path: Path):
-    """Adds new component to CMakeLists.txt"""
+    """ Adds new component or port to CMakeLists.txt"""
+    print("[INFO] Found CMakeLists.txt at '{}'".format(list_file))
+    with open(list_file, "r") as f:
+        lines = f.readlines()
+        index = 0
+        while "add_fprime_subdirectory" not in lines[index]:
+            index += 1
+        while "add_fprime_subdirectory" in lines[index]:
+            index += 1
+
+    if not confirm(
+        "Add component {} to {} {}?".format(
+            comp_path,
+            list_file,
+            "at end of file before topology inclusion?",
+        )
+    ):
+        return False
+
+    addition = 'add_fprime_subdirectory("${CMAKE_CURRENT_LIST_DIR}/' + str(comp_path) + '/")\n'
+    if addition in lines:
+        print("Component already added to CMakeLists.txt")
+        return True
+    lines.insert(index, addition)
+    with open(list_file, "w") as f:
+        f.write("".join(lines))
+    return True
+
+def regenerate(cmake_list_file):
+    #Purges and generates project for the user
+    currentDir = os.getcwd()
+    buildDir = (str(cmake_list_file).split("/"))[:-1]
+    os.chdir("/".join(buildDir))
+    try:
+        print("You may want to purge your project so you can regenerate:")
+        os.system("fprime-util purge")
+        os.system("fprime-util generate")
+    except:
+        os.system("fprime-util generate")
+    os.chdir(currentDir)
+
+
+def add_unit_tests(comp_path):
+    #Creates unit tests and moves them into test/ut directory
+    os.chdir(str(comp_path))
+    if confirm("Would you like to generate unit tests?: "):
+        os.system("fprime-util impl --ut")
```

Comment:
Instead of executing the currently running program by calling `os.system`, what about calling the `impl --ut` target from within python by calling the respective target?

Reply:
Got this working!